### PR TITLE
feat(signer) add a new groovy lib script to allow signing (only) the json files

### PR DESCRIPTION
**Pull request by dduportal**
Ref. jenkins-infra/helpdesk#2950

### CODE REVIEW

**Comment by dduportal** [comment]
@daniel-beck done, does it sounds good to you?

**Review by lemeurherve — APPROVED** [review]

File: lib/signer.groovy [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+list.each {
+  String file  = it.path
+  println "== ${file}"
+
+  // Load content of JSON file to be signed
+  File inputfile = new File(file)
+  String utf8Content = inputfile.getText("UTF-8")
+  if(utf8Content.length()<1) {
```
Comment (daniel-beck) [inline lib/signer.groovy:21]: `[]` is a really dangerous file to sign. Not strictly empty, but close enough. So this doesn't help.

Would we ever want to sign something with less than 100 or 200 bytes?

Comment (lemeurherve) [reply lib/signer.groovy:21]: Maybe too naive but how about testing the presence of `":` in utf8Content?

Comment (dduportal) [reply lib/signer.groovy:21]: ok, I've added 2 rules: it must contains a `: ` (as per @lemeurherve ) and the size should be more than 100 chars (as per @daniel-beck 's feedback).

Tested on the actual JSON files and it works
Tested with `[]`, `{}` (with and without JSONP) and fails as expected.

File: lib/signer.groovy [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    println "ERROR: the file ${file} is empty. Exiting."
+    System.exit(1)
+  }
+
+  // There is JSONp callback wrapping the envelope. Separator is the first comma (',').
+  int commaIndex = utf8Content.indexOf(',')
+  // Do not forget to remove the closing parenthses char at the end
+  String jsonBody = utf8Content.substring(commaIndex + 1, utf8Content.length()-1)
+  if(!jsonBody.contains(':') || jsonBody.length() < 100 ) {
+    println "ERROR: the JSON body in the ${file} is empty or with harmful content. Exiting."
```
[the remark this answers is not in the document]
Comment (dduportal) [reply lib/signer.groovy:31]: 🤔 make sense. WDYt @daniel-beck @wadeck ?

[the remark this answers is not in the document]
Comment (lemeurherve) [reply lib/signer.groovy:31]: I'd go for it if there is no objection from @jenkins-infra/security (comment also serving as a gentle ping)

[the remark this answers is not in the document]
Comment (daniel-beck) [reply lib/signer.groovy:31]: Leaving this explicit helps justify. Doesn't matter much either way though.

**Pull request merged**